modules/sdk-coin-sol/src/sol.ts

```diff
@@ -162,7 +168,56 @@ export class Sol extends BaseCoin {
       );
       const filteredOutputs = explainedTx.outputs.map((output) => _.pick(output, ['address', 'amount', 'tokenName']));
 
-      if (!_.isEqual(filteredOutputs, filteredRecipients)) {
+      if (filteredRecipients.length !== filteredOutputs.length) {
+        throw new Error('Number of tx outputs does not match with number of txParams recipients');
+      }
+
+      // For each recipient, check if it's a token tx (tokenName will exist if so)
+      // If it is a token tx, verify that the recipient address equals the derived address from explainedTx
+      // Derive the ATA if it is a native address and confirm it is equal to the explained tx recipient
+      const recipientChecks = await Promise.all(
+        filteredRecipients.map(async (recipientFromUser, index) => {
+          const recipientFromTx = filteredOutputs[index]; // This address should be an ATA
+
+          // Compare the amount string values because amount is (string | number)
+          const userAmountString =
+            typeof recipientFromUser.amount === 'string'
+              ? recipientFromUser.amount
+              : recipientFromUser.amount.toString();
+          const txAmountString =
+            typeof recipientFromTx.amount === 'string' ? recipientFromTx.amount : recipientFromTx.amount.toString();
```

Comment:
no need for this check.
both `number` and `string` have the `toString` function.

so you can do:
```
const userAmountString = recipientFromUser.amount.toString();
const txAmountString = recipientFromTx.amount.toString();
```